### ISSUE TITLE
Installing module fails when windows username has a space

### ISSUE THREAD

**Issue by rboy1** [luarocks/luarocks#1365]
### Please fill out the following:
- **Platform:**
  - Windows 10 64bit

- **LuaRocks version:**
  - `3.7.0`

Trying to install Luacheck using (from an Administrative command prompt)

> luarocks install luacheck

which results in the error:
> Error: Could not fetch rock file: Failed creating temporary directory luarocks-rock-luacheck-0.25.0-1: Failed setting permission exec for all

The windows username is "First Last" (i.e. has two words)

- **LuaRocks output from when the issue occurred:**
>os.execute:     icacls "C:\Users\FIRSTL~1\AppData\Local\Temp\luarocks_luarocks-rock-luacheck-0.25.0-1-3732" /inheritance:d /grant %USERNAME%:(F)
Invalid parameter "First"
Results: 3
  1 (nil): nil
  2 (string): exit
  3 (number): 87
>
>Error: Could not fetch rock file: Failed creating temporary directory luarocks-rock-luacheck-0.25.0-1: Failed setting permission exec for all

- **Description:**
If the windows username has a space (e.g. "First Last"), i.e. two or more words it fails. The `%USERNAME%` parameter should be encapsulated in quotes when passed to the command line.

**Comment by hishamhm** [comment]
Thanks for the report! @rboy1, can you confirm that this change fixes the behavior?

```diff
diff --git a/src/luarocks/fs/win32/tools.lua b/src/luarocks/fs/win32/tools.lua
index 5d98e52..3dfabc9 100644
--- a/src/luarocks/fs/win32/tools.lua
+++ b/src/luarocks/fs/win32/tools.lua
@@ -276,7 +276,7 @@ function tools.set_permissions(filename, mode, scope)
          return false, "Failed setting permission " .. mode .. " for " .. scope
       end
       -- Grant permissions available only to the current user
-      ok = fs.execute_quiet(vars.ICACLS .. " " .. fs.Q(filename) .. " /inheritance:d /grant %USERNAME%:" .. my_perms)
+      ok = fs.execute_quiet(vars.ICACLS .. " " .. fs.Q(filename) .. " /inheritance:d /grant \"%USERNAME%\":" .. my_perms)
       if not ok then
          return false, "Failed setting permission " .. mode .. " for " .. scope
       end
```

**Comment by rboy1** [comment]
Is there an easy way to build it on windows otherwise if you can get me a test build I can try it out

**Comment by hishamhm** [comment]
@rboy1 It should have been fixed on LuaRocks 3.8.0! Please give that version a try and reopen the bug if the problem persists! Thank you!

**Comment by rboy1** [comment]
Yes that seems to be resolved but I'm seeing another problem as described on this ticket:
https://github.com/luarocks/luarocks/issues/1287
